### DISCUSSION TITLE
pnpm install completes successfully without update to lock file

### DISCUSSION THREAD

**Discussion by kolaente** [renovatebot/renovate#24114] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

36.65.1

### If you're self-hosting Renovate, select which platform you are using.

Gitea or Forgejo

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

Renovate does not update the pnpm lockfile anymore. The logs say it did not need updating, but the CI fails to install dependencies because of an outdated lockfile. Checking out the PR locally and running `pnpm install` will also update the lockfile.

One example PR: https://kolaente.dev/vikunja/frontend/pulls/3725
CI run complainging about an outdated lockfile: https://drone.kolaente.de/vikunja/frontend/14650/1/3

Maybe similar to this: https://github.com/renovatebot/renovate/discussions/24043

Should a minimal reproduction repo be hosted on my Gitea instance or on Github? Given how this might be caused by self-hosting.

### Relevant debug logs

<details><summary>Logs</summary>

This is from a newly created PR:

```
DEBUG: Executing command (repository=vikunja/frontend, branch=renovate/axios-1.x)
       \"command\": \"docker run --rm --name=renovate_sidecar --label=renovate_child --user=0:0 -v \\\"/tmp/renovate/repos/gitea/vikunja/frontend\\\":\\\"/tmp/renovate/repos/gitea/vikunja/frontend\\\" -v \\\"/tmp/renovate/cache\\\":\\\"/tmp/renovate/cache\\\" -
CHE_DIR -w \\\"/tmp/renovate/repos/gitea/vikunja/frontend\\\" ghcr.io/containerbase/sidecar:9.19.3 bash -l -c \\\"install-tool node 18.17.1 && install-tool pnpm 8.7.0 && pnpm install --recursive --lockfile-only --ignore-scripts --ignore-pnpmfile\\\"\"" channel=
DEBUG: exec completed (repository=vikunja/frontend, branch=renovate/axios-1.x)
       \"durationMs\": 7295,
       \"stdout\": \"[15:06:44.434] INFO (10): Installing tool node v18.17.1...\\nv18.17.1\\n9.6.7\\n0.18.0\\n[15:06:47.232] INFO (10): Installed tool node in 2.7s.\\n[15:06:47.632] INFO (48): Installing tool pnpm v8.7.0...\\n8.7.0\\n[15:06:48.871] INFO (48): I
p/renovate/repos/gitea/vikunja/frontend\\\"\\n\",
       \"stderr\": \"\"
DEBUG: pnpm-lock.yaml hasn't changed (repository=vikunja/frontend, branch=renovate/axios-1.x)
DEBUG: No updated lock files in branch (repository=vikunja/frontend, branch=renovate/axios-1.x)
DEBUG: 1 file(s) to commit (repository=vikunja/frontend, branch=renovate/axios-1.x)
DEBUG: Preparing files for committing to branch renovate/axios-1.x (repository=vikunja/frontend, branch=renovate/axios-1.x)
DEBUG: git commit (repository=vikunja/frontend, branch=renovate/axios-1.x)
       \"deletedFiles\": [],
       \"ignoredFiles\": [],
       \"result\": {
         \"author\": null,
         \"branch\": \"renovate/axios-1.x\",
         \"commit\": \"eb1acffbf21ad84ecd7b74b208fcdd7397dc0dcc\",
         \"root\": false,
         \"summary\": {\"changes\": 1, \"insertions\": 1, \"deletions\": 1}
       }
DEBUG: Pushing refSpec renovate/axios-1.x:renovate/axios-1.x (repository=vikunja/frontend, branch=renovate/axios-1.x)
DEBUG: git push (repository=vikunja/frontend, branch=renovate/axios-1.x)
       \"result\": {
         \"pushed\": [
           {
             \"deleted\": false,
             \"tag\": false,
             \"branch\": true,
             \"new\": true,
             \"alreadyUpdated\": false,
             \"local\": \"refs/heads/renovate/axios-1.x\",
             \"remote\": \"refs/heads/renovate/axios-1.x\"
           }
         ],
         \"ref\": {\"local\": \"refs/remotes/origin/renovate/axios-1.x\"},
         \"remoteMessages\": {
           \"all\": [
             \"Create a new pull request for 'renovate/axios-1.x':\",
             \"https://kolaente.dev/vikunja/frontend/compare/main...renovate/axios-1.x\",
             \". Processing 1 references\",
             \"Processed 1 references in total\"
           ]
         }
       }
DEBUG: Setting current branch to main (repository=vikunja/frontend, branch=renovate/axios-1.x)
DEBUG: latest commit (repository=vikunja/frontend, branch=renovate/axios-1.x)
       \"branchName\": \"main\",
       \"latestCommitDate\": \"2023-08-27T15:00:38+00:00\"
 INFO: Branch created (repository=vikunja/frontend, branch=renovate/axios-1.x)
       \"commitSha\": \"eb1acffbf21ad84ecd7b74b208fcdd7397dc0dcc\"
DEBUG: Ensuring PR (repository=vikunja/frontend, branch=renovate/axios-1.x)
DEBUG: There are 0 errors and 0 warnings (repository=vikunja/frontend, branch=renovate/axios-1.x)
DEBUG: getBranchPr(renovate/axios-1.x) (repository=vikunja/frontend, branch=renovate/axios-1.x)
DEBUG: findPr(renovate/axios-1.x, undefined, open) (repository=vikunja/frontend, branch=renovate/axios-1.x)
DEBUG: getPrCache() (repository=vikunja/frontend, branch=renovate/axios-1.x)
DEBUG: Fetching changelog: https://github.com/axios/axios (1.4.0 -> 1.5.0) (repository=vikunja/frontend, branch=renovate/axios-1.x)
DEBUG: Creating PR (repository=vikunja/frontend, branch=renovate/axios-1.x)
       \"prTitle\": \"fix(deps): update dependency axios to v1.5.0\"
DEBUG: Creating pull request: fix(deps): update dependency axios to v1.5.0 (renovate/axios-1.x => main) (repository=vikunja/frontend, branch=renovate/axios-1.x)
DEBUG: lookupLabelByName(dependencies) (repository=vikunja/frontend, branch=renovate/axios-1.x)
DEBUG: Retrieved 0 org labels (repository=vikunja/frontend, branch=renovate/axios-1.x)
DEBUG: Retrieved 12 repo labels (repository=vikunja/frontend, branch=renovate/axios-1.x)
 INFO: PR created (repository=vikunja/frontend, branch=renovate/axios-1.x)
       \"pr\": 3725,
       \"prTitle\": \"fix(deps): update dependency axios to v1.5.0\"
DEBUG: addParticipants(pr=3725) (repository=vikunja/frontend, branch=renovate/axios-1.x)
DEBUG: setPrCache() (repository=vikunja/frontend, branch=renovate/axios-1.x)
DEBUG: Created Pull Request #3725 (repository=vikunja/frontend, branch=renovate/axios-1.x)
DEBUG: PR is not configured for automerge (repository=vikunja/frontend, branch=renovate/axios-1.x)
DEBUG: setBranchCommit() (repository=vikunja/frontend, branch=renovate/axios-1.x)
DEBUG: syncBranchState() (repository=vikunja/frontend, branch=renovate/ufo-1.x)
DEBUG: syncBranchState(): Branch cache not found, creating minimal branchState (repository=vikunja/frontend, branch=renovate/ufo-1.x)
```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by viceice** [comment]
I don't see any errors in your logs

**Reply by kolaente** [reply]
I can add more but the rest I check didn't really look different from this.

**Reply by rarkins** [reply]
You can see the pnpm command which is run. The before/after was unchanged for the lock file. First try running the exact same pnpm command and not just "pnpm install"

**Reply by kolaente** [reply]
Ran the exact same command, it changed the lockfile.

This was the command I ran: `pnpm install --recursive --lockfile-only --ignore-scripts --ignore-pnpmfile`.

**Reply by rarkins** [reply]
Please try to reproduce it locally first and then import the same to GitHub to see if it reproduces there

**Reply by kolaente** [reply]
You mean running the pnpm command locally or creating a minimal package.json in a folder and running renovate on top of that?

**Reply by rarkins** [reply]
I mean this:
- Get a simple reproduction repository which repeats the problem
- Verify that the same repository reproduces on github.com
- Then we see if we can get the same result too, if so we can debug it

**Reply by kolaente** [reply]
I've checked another project where I use renovate, and it seems it does not update any lockfile - not only pnpm. The other project uses composer and yarn and all PRs created for dependencies of it don't have a lockfile.

The composer one has an error message though:

```
Command failed: docker run --rm --name=renovate_sidecar --label=renovate_child --user=0:0 -v "/tmp/renovate/repos/gitea/kolaente/repo":"/tmp/renovate/repos/gitea/kolaente/repo" -v "/tmp/renovate/cache":"/tmp/renovate/cache" -e COMPOSER_CACHE_DIR -e COMPOSER_AUTH -e CONTAINERBASE_CACHE_DIR -w "/tmp/renovate/repos/gitea/kolaente/repo" ghcr.io/containerbase/sidecar:9.19.3 bash -l -c "install-tool php 8.2.9 && install-tool composer 2.5.8 && composer update laravel/framework:10.20.0 --with-dependencies --ignore-platform-req='ext-*' --ignore-platform-req='lib-*' --no-ansi --no-interaction --no-scripts --no-autoloader --no-plugins"
Composer could not find a composer.json file in /tmp/renovate/repos/gitea/kolaente/repo
To initialize a project, please create a composer.json file. See https://getcomposer.org/basic-usage
```

To me, this looks like the mount got lost somewhere along the way and it tried to update in an empty repo? Running `pnpm install` in an empty folder will not let pnpm exit with an error code, hence it might not get picked up as an error.

I'll set up a reproduction repo later today.

**Reply by rarkins** [reply]
Yes, it could be a mount problem. If you are using binarySource=docker then you need to make sure you're mapping the same host dir to the Renovate container.

**Comment by Zemnmez** [comment]
I'm seeing the same issue (my issue was closed in discussion #24046). still seems to be generating PRs with no lockfile update https://github.com/Zemnmez/monorepo/pull/3650

**Reply by rarkins** [reply]
@Zemnmez yours is not the same problem as described here, apart from the original over-general title. The problem here completes `pnpm install` successfully without any error, whereas yours has an error message which describe the problem (unmet peer dependencies) and is not a bug.

**Comment by Zemnmez** [comment]
> @Zemnmez yours is not the same problem as described here, apart from the lazy title. The problem here completes `pnpm install` successfully without any error, whereas yours has an error message which describe the problem (unmet peer dependencies) and is not a bug.

Gotcha, thanks!
